File: C++/Linked List Cycle.cpp
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    /**
     * @param head: The first node of linked list.
     * @return: True if it has a cycle, or false
     */
    bool hasCycle(ListNode *head) {
        // write your code here
        if (head == NULL || head ->next == NULL) return false;
        ListNode *node1 = head, *node2 = head;
        node1 = node1->next;
        node2 = (node2->next)->next;
        if (node2 == NULL) return false;
        while (node1 != node2) {
            node1 = node1->next;
            node2 = node2->next;
            if (node2 == NULL) return false;
            node2 = node2->next;
            if (node2 == NULL) return false;
        }
        return true;
    }
};

File: Python/Linked List Cycle.py
# @iFighting

# Definition for singly-linked list.
class ListNode:
     def __init__(self, x):
         self.val = x
         self.next = None

class Solution:
    # @param head, a ListNode
    # @return a boolean
    def hasCycle(self, head):
        node1=head
        node2=head
        while node2 != None:
            node1=node1.next
            node2=node2.next
            if node2 == None:
                return False
            node2=node2.next;
            if(node2==node1):
                return True
        return False